Upgrade application to rails 6.0.0

ThuTrinh opened the pull request:
Upgrade application to rails 6.0.0

## Description
- Updated rails gem and a few related dependency gems
- Upgrade application to rails 6.0.0 by running `rails app:update` 

I ran the application locally and all seems fine 🤞🏼 

- [ ] mtest test/graphql/loaders/association_loader_test.rb
- [ ] mtest test/graphql/loaders/event_hours_loader_test.rb
- [ ] mtest test/graphql/loaders/signup_count_loader_test.rb
- [ ] mtest test/graphql/resolvers/event_resolver_test.rb
- [ ] mtest test/graphql/resolvers/user_resolver_test.rb
- [ ] mtest test/graphql/resolvers/volunteer_resolver_test.rb (@kerrisim )
- [ ] mtest test/models/event_test.rb (@kayison )
- [ ] mtest test/models/organization_test.rb (@kayison )

## CCs
@zendesk/volunteer

@samzx @kerrisim and @kayison @craig-day - would you mind having a scan of the new configs files that came with rails 6.0? I am not sure they are all correct.

## Risks (if any)
* [HIGH] - rails upgrade to 6.0.0 with a bunch of changes to how things are configured. not sure if this works in stg or prod 😨

ThuTrinh committed:
update gems

ThuTrinh committed:
update gems

ThuTrinh committed:
apparently rails 6.0 is more stable :shrug:

ThuTrinh committed:
apparently rails 6.0 is more stable :shrug:

ThuTrinh committed:
upgrade app to rails 6.0

ThuTrinh committed:
exclude node_modules from rubocop

kayison committed:
Remove non-precompiled nokogiri

samzx reviewed (commented):
I had to do
```
mkdir tmp
mkdir tmp/pids
```
before starting, other than that, it seems to work fine!

Also looks like there's lots of tests to fix 😢 

Don't know much about the config files unfortunately, but they seem fine!

kayison committed:
Manually add nokogiri 1.11.1

kayison committed:
Downgrade nokogiri 1.11.1 to 1.10.10

- Heroku is not able to find either nokogiri gemfile and the build is failing
- Try to go back to older version to get Heroku build working again

dependabot-preview[bot] committed:
Bump webmock from 3.11.0 to 3.11.2

Bumps [webmock](https://github.com/bblimke/webmock) from 3.11.0 to 3.11.2.
- [Release notes](https://github.com/bblimke/webmock/releases)
- [Changelog](https://github.com/bblimke/webmock/blob/master/CHANGELOG.md)
- [Commits](https://github.com/bblimke/webmock/compare/v3.11.0...v3.11.2)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

kayison committed:
Add vendor/cache gems

kayison committed:
Revert "Merge pull request #516 from zendesk/dependabot/bundler/omniauth-saml-2.0.0"

This reverts commit adcef156241a5f45613b686af1d2c5898f473e08, reversing
changes made to c6834215a44e12f7a9af97c307fe4f0a88d5546b.

ThuTrinh committed:
rebased master

ThuTrinh committed:
fix rebase issues

ThuTrinh commented:
I am not too sure about this theory since I am not well versed in Ruby or Rails, I will need help to validate my theory.

Since the only changes are framework and dependency version updates, it is likely that the test are failing due to some breaking change Rails 6 has imposed on how ActiveRecord scope works?

From what I can tell, it looks like the tests are failing due to unexpected result from filtering queries. It looks like we are using `scope` to apply the filtering.

This [thread](https://github.com/Shopify/graphql-batch/issues/77#issuecomment-356964474) suggests we should not use the `scope` argument of `ActiveRecord::Associations::Preloader#preload` and warns that future internal refactoring of rails could break this undocumented feature.

I also found these thread discussing fixes for scope chaining no longer leaking into the class level. I don't quite understand if its related, will need some assistance to dig deeper.
- https://github.com/rails/rails/pull/32380
- https://github.com/rails/rails/pull/35280

cc/ @kerrisim @kayison

ThuTrinh commented:
This is a massive PR with lots of change. Closing it in favour of smaller PRs that will address a smaller scope of the upgrade.
